crates/solver/src/interactions/allowances.rs

```diff
@@ -43,26 +49,26 @@ pub trait AllowanceManaging: Send + Sync {
 
 #[derive(Debug, Eq, PartialEq)]
 pub struct ApprovalRequest {
-    pub token: H160,
-    pub spender: H160,
-    pub amount: U256,
+    pub token: Address,
+    pub spender: Address,
+    pub amount: alloy::primitives::U256,
```

Comment:
Seems like some `U256` conversions snuck in as well. Fine to keep it I just found it curious that (presumably) AI did that.

Reply:
Nope, all structure migrations were me. I explicitly forced it *not* to do these. I just didn't update the description, will do